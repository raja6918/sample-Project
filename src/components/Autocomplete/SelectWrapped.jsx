import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import { Creatable } from 'react-select';
import Typography from '@material-ui/core/Typography';
import Icon from '@material-ui/core/Icon';
import SierraTooltip from '../../_shared/components/SierraTooltip';

import Option from './Option';

class SelectWrapper extends Component {
  render() {
    const { createComponent, createLabel, t, isToolTip, ...other } = this.props;

    if (createComponent) {
      return (
        <Creatable
          optionComponent={Option}
          promptTextCreator={value =>
            `${
              createLabel ? createLabel : t('GLOBAL.form.capCreate')
            } '${value}'`
          }
          arrowRenderer={arrowProps => {
            return arrowProps.isOpen ? (
              <Icon>arrow_drop_up</Icon>
            ) : (
              <Icon>arrow_drop_down</Icon>
            );
          }}
          clearRenderer={() => <Icon>clear</Icon>}
          valueComponent={valueProps => {
            const { children } = valueProps;

            return <div>{children}</div>;
          }}
          {...other}
        />
      );
    } else {
      return (
        <Select
          optionComponent={Option}
          noResultsText={
            <Typography>{t('GLOBAL.dataNotFound.noResults')}</Typography>
          }
          arrowRenderer={arrowProps => {
            return arrowProps.isOpen ? (
              <Icon>arrow_drop_up</Icon>
            ) : (
              <Icon>arrow_drop_down</Icon>
            );
          }}
          clearRenderer={() => <Icon>clear</Icon>}
          valueComponent={valueProps => {
            const { children } = valueProps;

            return isToolTip ? (
              <SierraTooltip
                position="bottom"
                html={<p style={{ padding: '20px' }}>{children}</p>}
              >
                <div>{`${children.substr(0, 30)}...`}</div>
              </SierraTooltip>
            ) : (
              <div>{children}</div>
            );
          }}
          {...other}
        />
      );
    }
  }
}

SelectWrapper.propTypes = {
  createComponent: PropTypes.bool.isRequired,
  createLabel: PropTypes.string,
  t: PropTypes.func.isRequired,
  isToolTip: PropTypes.bool,
};

SelectWrapper.defaultProps = {
  createLabel: '',
  isToolTip: false,
};
export default SelectWrapper;
