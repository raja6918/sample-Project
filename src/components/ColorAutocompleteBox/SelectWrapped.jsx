import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select, { components } from 'react-select3';
import Icon from '@material-ui/core/Icon';

import './style.scss';
import 'react-perfect-scrollbar/dist/css/styles.css';

class SelectWrapper extends Component {
  DropdownIndicator = props => {
    return (
      components.DropdownIndicator && (
        <components.DropdownIndicator {...props}>
          <Icon>arrow_drop_down</Icon>
        </components.DropdownIndicator>
      )
    );
  };

  CustomOption = props => {
    return (
      <div
        style={{
          fontWeight: props.data.primary ? 500 : 400,
          color: props.data.primary ? this.props.color : '',
        }}
        className={props.data.primaryLast ? 'margin-check' : ''}
      >
        <components.Option {...props} />
      </div>
    );
  };

  render() {
    const {
      value,
      onChange,
      options,
      disabled,
      placeholder,
      color,
    } = this.props;
    const selectedOption = Array.isArray(options)
      ? options.find(option => option.value === value)
      : '';
    return (
      <Select
        styles={{
          menu: styles => ({
            ...styles,
            maxHeight: '240px',
          }),
          menuList: styles => ({
            ...styles,
            maxHeight: '240px',
          }),
          indicatorContainer: styles => ({ ...styles, padding: 0 }),
          dropdownIndicator: styles => ({
            ...styles,
            color: disabled ? '#cccccc !important' : '',
          }),
          control: styles => ({
            ...styles,
            border: 0,
            borderBottom: '1px solid #cccccc',
            borderRadius: 'unset',
            background: 'transparent',
            boxShadow: 0,
            cursor: 'pointer',
          }),
          option: (styles, { data, isSelected }) => ({
            ...styles,
            background: isSelected ? 'rgba(0, 0, 0, 0.04)' : 'transparent',
            color: data.primary ? color : '#000000',
            cursor: 'pointer',
            '&:hover': {
              background: 'rgba(0, 0, 0, 0.04)',
            },
          }),
          placeholder: styles => ({
            ...styles,
            color: disabled ? '#cccccc !important' : '',
          }),
          singleValue: (styles, { data }) => ({
            ...styles,
            color: data.primary ? color : '',
          }),
        }}
        isClearable
        components={{
          Option: this.CustomOption,
          DropdownIndicator: this.DropdownIndicator,
          IndicatorSeparator: () => null,
        }}
        value={selectedOption || ''}
        onChange={onChange}
        options={options}
        isDisabled={disabled}
        placeholder={placeholder}
      />
    );
  }
}

SelectWrapper.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.boolean,
    PropTypes.string,
    PropTypes.number,
  ]),
  placeholder: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([
        PropTypes.boolean,
        PropTypes.string,
        PropTypes.number,
      ]).isRequired,
      primary: PropTypes.boolean,
      primaryLast: PropTypes.boolean,
    })
  ).isRequired,
  onChange: PropTypes.func.isRequired,
  color: PropTypes.string,
  disabled: PropTypes.bool,
  defaultValue: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
    PropTypes.null,
  ]),
};

SelectWrapper.defaultProps = {
  placeholder: '',
  value: '',
  color: '#000000AB',
  disabled: false,
  defaultValue: null,
};
export default SelectWrapper;
