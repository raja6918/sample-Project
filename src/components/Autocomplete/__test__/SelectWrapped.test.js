import React from 'react';
import { shallow } from 'enzyme';

import SelectWrapped from '../SelectWrapped';

const templateOptions = [
  { value: 1, label: 'Tempalte 1' },
  { value: 2, label: 'Tempalte 2' },
  { value: 3, label: 'Tempalte 3' },
];
const categoryOptions = [
  { value: 'Category 1', label: 'Category 1' },
  { value: 'Category 2', label: 'Category 2' },
  { value: 'Category 3', label: 'Category 3' },
];
test.skip('SelectWrapped component renders', () => {
  const wrapperRegular = shallow(
    <SelectWrapped
      createComponent={false}
      t={jest.fn()}
      instanceId={'sourceTemplate'}
      name={'sourceTemplate'}
      options={templateOptions}
      placeholder={'Source Template *'}
    />
  );

  expect(wrapperRegular).toMatchSnapshot();

  const wrapperCreate = shallow(
    <SelectWrapped
      createComponent={true}
      t={jest.fn()}
      instanceId={'category'}
      name={'category'}
      options={categoryOptions}
      placeholder={'Category'}
    />
  );

  expect(wrapperCreate).toMatchSnapshot();
});
