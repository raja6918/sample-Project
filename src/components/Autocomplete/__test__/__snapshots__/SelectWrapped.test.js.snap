// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectWrapped component renders 1`] = `
<Select
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={true}
  closeOnSelect={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  instanceId="sourceTemplate"
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  name="sourceTemplate"
  noResultsText={<UNDEFINED />}
  onBlurResetsInput={true}
  onCloseResetsInput={true}
  onSelectResetsInput={true}
  openOnClick={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "label": "Tempalte 1",
        "value": 1,
      },
      Object {
        "label": "Tempalte 2",
        "value": 2,
      },
      Object {
        "label": "Tempalte 3",
        "value": 3,
      },
    ]
  }
  pageSize={5}
  placeholder="Source Template *"
  removeSelected={true}
  required={false}
  rtl={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  trimFilter={true}
  valueComponent={[Function]}
  valueKey="value"
/>
`;

exports[`SelectWrapped component renders 2`] = `
<CreatableSelect
  arrowRenderer={[Function]}
  clearRenderer={[Function]}
  filterOptions={[Function]}
  instanceId="category"
  isOptionUnique={[Function]}
  isValidNewOption={[Function]}
  menuRenderer={[Function]}
  name="category"
  newOptionCreator={[Function]}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "label": "Category 1",
        "value": "Category 1",
      },
      Object {
        "label": "Category 2",
        "value": "Category 2",
      },
      Object {
        "label": "Category 3",
        "value": "Category 3",
      },
    ]
  }
  placeholder="Category"
  promptTextCreator={[Function]}
  shouldKeyDownEventCreateNewOption={[Function]}
  showNewOptionAtTop={true}
  valueComponent={[Function]}
/>
`;
